todo-list-app/routers/users.py

```diff
@@ -0,0 +1,128 @@
+    user = await crud_get_existing_user(user_id=id_, db=db)
+    result = UserDataOutput(message="User was found.", user=user)
+    return result
+
+
+@router.put("/{id_}", response_model=UserDataOutput)
+async def update_user(id_: int, user_data: UserUpdate, db: AsyncSession = Depends(get_db),
+                      requester_id: int = Depends(get_user_id),
+                      requester_role: str = Depends(get_user_role)):
+    """
+    Endpoint to update an existing user.
+    
+    Returns:
+       Returns the new changes to the user.
+    """
+    user = await crud_update_user(user_id=id_, requester_id=requester_id,
+                                  requester_role=requester_role, user_data=user_data, db=db)
+    result = UserDataOutput(message="User was updated.", user=user)
+    return result
+
+
+@router.delete("/{id_}", response_model=ConnectionResponse)
+async def delete_user(id_: int, db: AsyncSession = Depends(get_db),
+                      requester_id: int = Depends(get_user_id),
+                      requester_role: str = Depends(get_user_role)):
+    """
+    Endpoint to delete an existing user.
+    
+    Returns:
+       Returns info about the transaction.
+    """
+    result = await crud_delete_user(user_id=id_, requester_id=requester_id,
+                                    requester_role=requester_role, db=db)
+    return result
+
+
+@router.patch("/{id_}", response_model=ConnectionResponse)
+async def set_role(id_: int, new_role: NewRole, db: AsyncSession = Depends(get_db),
+                   requester_role: str = Depends(get_user_role)):
+    """
+    Endpoint to change the role of an existing user.
+    
+    Returns:
+       Returns info about the transaction.
+    """
+    result = await crud_set_role(user_id=id_, new_role=new_role,
+                                 requester_role=requester_role, db=db)
+    return result
+
+
+@router.get("/")
+async def get_all_users(db: AsyncSession = Depends(get_db),
+                        user_role: str = Depends(get_user_role)) -> list:
+    """
+    Endpoint to get all existing users.
+    
+    Returns:
+       Returns a list of all users from the Users table.
+    """
+    result = await crud_get_all_users(user_role=user_role, db=db)
+    return result
+
+
+@router.post("/reminders", status_code=200, response_model=ConnectionResponse)
+async def set_daily_reminder(reminder: DailyReminder,
+                             db: AsyncSession = Depends(get_db),
+                             user_id: int = Depends(get_user_id)):
+    """
+    Endpoint to set the daily reminders.
+
+    Returns:
+        Returns info about the transaction.
+    """
+    reminder = reminder.model_dump()['reminder']
+    result = await set_reminder(user_id=user_id, reminder_value=reminder, db=db)
+    return result
+
+
+@router.post("/send_reminders", status_code=200, response_model=ConnectionResponse)
+async def send_daily_reminder(job_timezone: TimeZoneInput, db: AsyncSession = Depends(get_db),
```

Comment:
I we tried to send messages, but got this error

```
curl -X 'POST' \
  'http://localhost:8000/api/v1/users/send_reminders' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer <token>' \
  -H 'Content-Type: application/json' \
  -d '{
  "timezone": "UTC"
}'
```

```
{
  "detail": "An internal server error occurred. Please try again later."
}
```


Could you please add traceback in logging so it`ll be easier to debug problems?

Currently it not really informative
```
todo-list-app-1      | INFO:     127.0.0.1:54810 - "POST /api/v1/users/send_reminders HTTP/1.1" 500 Internal Server Error
```